network-config-analyzer/IstioPolicyYamlParser.py

```diff
@@ -74,6 +74,41 @@ def parse_ns_str(self, ns):
         ns_obj = self.peer_container.get_namespace(ns)
         return self.peer_container.get_namespace_pods(ns_obj)
 
+    def parse_principal_str(self, principal):
+        """
+        parse a principal string from source component in rule
+        :param principal: the principal string, currently assuming in format: "cluster.local/ns/<ns-str>/sa/<sa-str>"
+        :return: PeerSet: All pods with the given ns + sa_name as extracted from principal str
+        """
+        if '*' in principal:
+            self.syntax_error(
+                'error parsing principal regular expr: currently not supporting prefix/suffix patterns ', principal)
+        # principal_str_example = "cluster.local/ns/default/sa/sleep"
+        # TODO: support a more general pattern for principal str (prefix by istio trust-domain)
+        # TODO: tighter checks in parsing the principal str
+        principal_pattern = 'cluster.local/ns/([\w-]+)/sa/([\w-]+)'
```

Comment:
The `\w` here gives me many warnings. I think it should be `\\w` or you should use Python's [raw string notation](https://docs.python.org/3/library/re.html#raw-string-notation).

Reply:
done